lib/JSON/Validator.pm

```diff
@@ -258,7 +258,10 @@ sub _resolve_schema {
 
         # int($v) returns the memory address of the reference. The "if" below
         # will not resolve the same "$ref" over again.
-        next if ref $v and $self->{seen}{int($v)}++;
+        next if ref $v and $self->{seen}{int($v)};
```

Comment:
That's a really interesting alternative as it removes the reliance on memory addresses entirely.

I'm still interested to see how we manage to get into this position in the first place.. that still perplexes me :(